staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics_test.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/component-base/metrics/testutil"
+)
+
+func TestRecordKMSOperationLatency(t *testing.T) {
+	testCases := []struct {
+		name         string
+		methodName   string
+		duration     time.Duration
+		operationErr error
+		want         string
+	}{
+		{
+			name:         "operation success",
+			methodName:   "/v2alpha1.KeyManagementService/Encrypt",
+			duration:     1 * time.Second,
+			operationErr: nil,
```

Comment:
You are missing the `"unknown-non-grpc"` error case (see the diff from earlier).